### PR TITLE
[RFC] statusline improvements

### PR DESCRIPTION
**Pull request by blueyed**
Fixes https://github.com/neomake/neomake/issues/915.

### CODE REVIEW

File: autoload/neomake/statusline.vim [hunk 1]
```diff
@@ -89,3 +91,143 @@ endfunction
...
+    if has_key(s:cache, a:bufnr)
+        unlet s:cache[a:bufnr]
+    endif
+endfunction
+
+augroup neomake_statusline
+    autocmd!
+    autocmd User NeomakeJobStarted,NeomakeJobFinished call s:clear_cache(g:neomake_hook_context.jobinfo.bufnr)
+    " Trigger redraw of all statuslines.
+    autocmd User NeomakeJobFinished let &stl = &stl
```
Comment (wsdjeg) [inline autoload/neomake/statusline.vim]: why not use : 
```
autocmd User NeomakeJobFinished redrawstatus!
```

**Comment by blueyed** [comment]
Need to finish docs, but would appreaciate any feedback already.

/cc @chrisbra @rdnetto regarding integration into vim-airline

**Comment by chrisbra** [comment]
I am missing what kind of airline problem you are trying to solve here.

**Comment by blueyed** [comment]
@chrisbra 
See your comment at https://github.com/vim-airline/vim-airline/pull/1275#issuecomment-273039607 - that's basically it.

It needs adoption in Airline of course, and that's why I ask for feedback.

**Comment by chrisbra** [comment]
I am still missing something. Is this about getting the counts from the quickfix list (instead of the location list, which seems to be currently the only supported way by the airline neomake extension?) If so, can you please point me to a specific commit, that only does that? I see to much neomake specific stuff in your commits and feel it hard to find the relevant code. I could then view a diff with the current neomake extension, which allows me to better review. Or is this about something else?

**Comment by blueyed** [comment]
@chrisbra 
Airline could just use the new `neomake#statusline#get()`.
See the doc added in the diff (WIP).

**Comment by blueyed** [comment]
:fireworks:

**Pull request closed without merging**